scope.go

```diff
@@ -209,48 +224,78 @@ func newRootScope(opts ScopeOptions, interval time.Duration) *scope {
 }
 
 // report dumps all aggregated stats into the reporter. Should be called automatically by the root scope periodically.
-func (s *scope) report(r StatsReporter) {
+func (s *scope) report(r StatsReporter) (reported bool) {
 	s.cm.RLock()
 	for name, counter := range s.counters {
-		counter.report(s.fullyQualifiedName(name), s.tags, r)
+		if rep := counter.report(s.fullyQualifiedName(name), s.tags, r); rep {
+			reported = true
+		}
 	}
 	s.cm.RUnlock()
 
 	s.gm.RLock()
 	for name, gauge := range s.gauges {
-		gauge.report(s.fullyQualifiedName(name), s.tags, r)
+		if rep := gauge.report(s.fullyQualifiedName(name), s.tags, r); rep {
+			reported = true
+		}
 	}
 	s.gm.RUnlock()
 
 	// we do nothing for timers here because timers report directly to ths StatsReporter without buffering
+	s.tm.RLock()
+	for _, timer := range s.timersSlice {
+		if rep := timer.hasReported(); rep {
+			reported = true
+		}
+	}
+	s.tm.RUnlock()
 
 	s.hm.RLock()
 	for name, histogram := range s.histograms {
-		histogram.report(s.fullyQualifiedName(name), s.tags, r)
+		if rep := histogram.report(s.fullyQualifiedName(name), s.tags, r); rep {
+			reported = true
+		}
 	}
 	s.hm.RUnlock()
+
+	return
 }
 
-func (s *scope) cachedReport() {
+func (s *scope) cachedReport() (reported bool) {
 	s.cm.RLock()
 	for _, counter := range s.countersSlice {
-		counter.cachedReport()
+		if rep := counter.cachedReport(); rep {
+			reported = true
+		}
```

Comment:
`reported = reported || counter.cachedReport()` ?

Reply:
The report calls are required and stateful, so we'd need to do it the other way around, but yeah, I like the conciseness. Thanks!